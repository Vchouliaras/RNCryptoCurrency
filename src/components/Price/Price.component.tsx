import * as React from 'react'
import { Title } from '@shoutem/ui'
import { View, Text } from 'react-native'

import { IPriceProps } from '../../types'

const Price: React.SFC<IPriceProps> = ({ price = 0, symbol }) => (
  <View>
    <Title>
      <Text>{symbol}</Text>
      {price && price.toFixed(2)}
    </Title>
  </View>
)

export default Price
