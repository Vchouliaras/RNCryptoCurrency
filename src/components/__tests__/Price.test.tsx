import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Price from '../Price/Price.component'

describe('Price Component', () => {
  it('renders correctly', () => {
    const wrapper: ShallowWrapper = shallow(<Price price={12} symbol={'$'} />)
    expect(wrapper).toMatchSnapshot()
  })
})
