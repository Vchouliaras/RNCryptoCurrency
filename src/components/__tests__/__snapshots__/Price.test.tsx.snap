// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Component renders correctly 1`] = `
<View>
  <Styled(Animated(Text))>
    <Text>
      $
    </Text>
    12.00
  </Styled(Animated(Text))>
</View>
`;
